src/main/java/org/elasticsearch/hadoop/mr/EsInputFormat.java

```diff
@@ -0,0 +1,456 @@
+import org.elasticsearch.hadoop.serialization.ScrollReader;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * ElasticSearch {@link InputFormat} for streaming data (typically based on a query) from ElasticSearch.
+ * Returns the document ID as key and its content as value.
+ *
+ * <p/>This class implements both the "old" (<tt>org.apache.hadoop.mapred</tt>) and the
+ * "new" (<tt>org.apache.hadoop.mapreduce</tt>) API.
+ * @param <K> the type of the key
+ * @param <V> the type of the value
+ */
+public class EsInputFormat<K, V> extends InputFormat<K, V> implements org.apache.hadoop.mapred.InputFormat<K, V> {
+
+  private static Log log = LogFactory.getLog(EsInputFormat.class);
+
+  /**
+   * Input Split
+   */
+  protected static class EsInputSplit extends InputSplit implements org.apache.hadoop.mapred.InputSplit {
+    private PartitionDefinition partition;
+
+    public EsInputSplit() {}
+
+    public EsInputSplit(PartitionDefinition partition) {
+      this.partition = partition;
+    }
+
+    @Override
+    public long getLength() {
+      // TODO: can this be computed easily?
+      return 1L;
+    }
+
+    @Override
+    public String[] getLocations() {
+      return partition.getHostNames();
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+      partition.write(out);
+    }
+
+    @Override
+    public void readFields(DataInput in) throws IOException {
+      partition = new PartitionDefinition(in);
+    }
+
+    public PartitionDefinition getPartition() {
+      return partition;
+    }
+  }
+
+  /**
+   *
+   * @param <K>
+   * @param <V>
+   */
+  protected abstract static class EsInputRecordReader<K, V>
+    extends RecordReader<K, V> implements org.apache.hadoop.mapred.RecordReader<K, V> {
+
+    private int read = 0;
+    private EsInputSplit esSplit;
+    private ScrollReader scrollReader;
+
+    private RestRepository client;
+    private SearchRequestBuilder queryBuilder;
+    private ScrollQuery scrollQuery;
+
+    // reuse objects
+    private K currentKey;
+    private V currentValue;
+
+    private long size = 0;
+
+    private HeartBeat beat;
+    private Progressable progressable;
+
+    // default constructor used by the NEW api
+    public EsInputRecordReader() {
+    }
+
+    // constructor used by the old API
+    public EsInputRecordReader(org.apache.hadoop.mapred.InputSplit split, Configuration job, Reporter reporter) {
+      reporter.setStatus(split.toString());
+      init((EsInputSplit) split, job, reporter);
+    }
+
+    // new API init call
+    @Override
+    public void initialize(InputSplit split, TaskAttemptContext context) throws IOException {
+      // org.elasticsearch.hadoop.mr.compat.TaskAttemptContext compatContext
+      // = CompatHandler.taskAttemptContext(context);
+      // compatContext.setStatus(split.toString());
```

Comment:
This is changed. Removed CompatHandler usage to get the context. Rather using context passed to the method as parameter.